lib/saml_idp/xml_security.rb

```diff
@@ -81,12 +81,14 @@ def signature_algorithm(options)
           algorithm(options[:get_params][:SigAlg])
         else
           ref_elem = REXML::XPath.first(self, "//ds:Reference", {"ds"=>DSIG})
+          return nil unless ref_elem
           algorithm(REXML::XPath.first(ref_elem, "//ds:DigestMethod", {"ds"=>DSIG}))
         end
       end
 
       def fingerprint_cert(cert, options)
         digest_algorithm = signature_algorithm(options)
+        return nil unless digest_algorithm
         digest_algorithm.hexdigest(cert.to_der)
```

Comment:
Instead of the early return could we use nil-safe navigator here?
```suggestion
        digest_algorithm&.hexdigest(cert.to_der)
```